ethcore/light/src/client/header_chain.rs

```diff
@@ -230,7 +390,7 @@ impl HeaderChain {
 					best.hash
 				};
 
-				self.headers.read().get(&hash).cloned()
+				load_from_db(hash)
```

Comment:
The DB should do this automatically, right? I plan to add customized caching in the future but I think it's beyond the scope of this PR.